### ISSUE TITLE
Apply some default for hr tags

### ISSUE THREAD

**Issue by pskim** [18F/web-design-standards#892]
The default browser style for hr tags doesn't fit well with the visual design of the framework. Possible solution is to style the hr tag or provide another method of a visual separator.

<img width="1035" alt="screen shot 2015-11-18 at 8 36 39 pm" src="https://cloud.githubusercontent.com/assets/14101848/11262468/7ca7cbb6-8e34-11e5-8a42-76a013ec9092.png">

**Comment by maya** [comment]
@pskim thanks for your feedback! That's a great idea, hopefully we can design one durring our next phase of work.

**Comment by thecristen** [comment]
This got me thinking about some possible designs.

With `$gray-cool-light`: 
![image](https://cloud.githubusercontent.com/assets/2136286/11543580/889d9496-990b-11e5-81f8-8579963ed3af.png)

And `$gray-warm-light`: 
![image](https://cloud.githubusercontent.com/assets/2136286/11543610/a20823d8-990b-11e5-8771-3b08a5f6b4df.png)

A bit thinner now: 
![image](https://cloud.githubusercontent.com/assets/2136286/11543623/b5095826-990b-11e5-8b64-39090e193a16.png)

(Codepen if anyone wants to play: http://codepen.io/thecristen/pen/dGbJjP)

**Comment by maya** [comment]
Style I've used on a [previous project](https://federalist-proxy.app.cloud.gov/preview/18f/clinical-trials-prototype/clean/study-detail/):
<img width="793" alt="screen shot 2017-10-24 at 11 35 18 am" src="https://user-images.githubusercontent.com/5249443/31961454-a6d9a1aa-b8af-11e7-98f3-770b7c84c6e9.png">

Also, 18F site:

<img width="694" alt="screen shot 2017-10-24 at 11 38 34 am" src="https://user-images.githubusercontent.com/5249443/31961573-efd0dbd0-b8af-11e7-8b64-5637e0181afa.png">

**Comment by thisisdano** [comment]
There are some great suggestions in here, and I think it's a good idea to provide some styling for the `hr` element. My primary concern with this is that we begin to limit the scope of styling we apply to base elements.

We're learning that styling base elements causes problems when developers try to integrate the standards into an existing codebase — for instance, the USWDS styling of an `h2` would very likely clash with a pre-existing definition of `h2`. To make our code more interoperable, we're learning to reduce this base element styling.

How to do this?

I like [David Bushell's method](https://dbushell.com/2012/04/18/scoping-typography-css/) of scoping base element styling to a content container — in our case, to `usa-content`. Scoping our base element styling to something like `.usa-content hr` allows reliable styling from CMS output without affecting application styling outside the content container. CMS output is also a good place to support `hr` styling because styling options can be very limited in CMS content.

Outside the content container, it's less clear that using `hr`s is a good idea. If we're interested in adding lines/rules, it's usually a better and more consistent to add a border to an existing element's box styling. I'm comfortable adding a `.usa-line` class rule we can apply to `hr`s, but I don't think it should be a common solution. For that reason, I'm disinclined to build out a full suite of variants for `usa-line` (like `usa-line-thick` or `usa-line-primary`). Let's add a sensible default, with a practical practical scope, but leave it minimal for now.

// @maya @donjo thoughts?

**Comment by donjo** [comment]
I think I prefer the `.usa-line` class method over the `.usa-content hr` style, at least initially, since I don't think we've been very explicit about what happens to things inside `.usa-content` and it's kind of a class you discover once you start copying template code and wondering where paragraphs got their margins. 

Maybe create the standalone class rule first and then think about whether there are any other things that should be brought under the `.usa-content` scope and no longer be globally defined (or have different definitions when they are part of a content block) and add a default `hr` rule at that time.

**Comment by thisisdano** [comment]
Yeah, I'm comfortable with that. Thanks. The `usa-content` scope issue is a nontrivial one for later.

**Comment by maya** [comment]
I think a sensible default sounds good and a specialized class too. We can always add `usa-content hr` in the future too. In my experience, I've needed hr's in components (non `usa-content`) also. 

A note on naming: we should think through how we name things a bit. Would `usa-hr` make more sense bc as a dev I'd be writing `<hr class="usa-hr">`? Kind of like buttons are `<button class=usa-button></button>`. Or is something like `usa-line` more explicit. We might want to research other libraries to how they name this component too to get ideas. Also, consider spacing, in my last project I had a need for generous spacing (60px) for written content and tighter spacing 15px-20px for a sidebar filter component.

**Comment by thisisdano** [comment]
I think I like `usa-hr` since it shouldn't be used on anything _but_ an `hr`.
